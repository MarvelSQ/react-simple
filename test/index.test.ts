import * as babel from '@babel/core';
import ReactSimple from '../src';

const simpleCode = `
function App() {
  let state = 1;
  let enable = true;
  function handleAdd() {
    const enable = false;
    enable = !enable;
    state = state + 1;
  }
  return <div onClick={handleAdd}>{state}</div>;
}`;

describe('simple', () => {
  test('let state', () => {
    const result = babel.transformSync(simpleCode, {
      presets: ['@babel/preset-react'],
      plugins: [ReactSimple],
    });

    expect(result?.code).toMatchSnapshot();
  });
});
