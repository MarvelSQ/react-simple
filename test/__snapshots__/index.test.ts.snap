// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple let state 1`] = `
"function App() {
  const [state, setState] = 1;
  const [enable, setEnable] = true;
  const handleAdd = useCallback(() => {
    const enable = false;
    enable = !enable;
    setState(state + 1);
  }, [state]);
  return /*#__PURE__*/React.createElement(\\"div\\", {
    onClick: handleAdd
  }, state);
}"
`;
